backend/danswer/indexing/indexing_pipeline.py

```diff
@@ -113,20 +124,59 @@ def get_doc_ids_to_update(
     return updatable_docs
 
 
-@log_function_time()
-def index_doc_batch(
+def index_doc_batch_with_handler(
     *,
     chunker: Chunker,
     embedder: IndexingEmbedder,
     document_index: DocumentIndex,
     document_batch: list[Document],
     index_attempt_metadata: IndexAttemptMetadata,
+    attempt_id: int | None,
     db_session: Session,
     ignore_time_skip: bool = False,
 ) -> tuple[int, int]:
-    """Takes different pieces of the indexing pipeline and applies it to a batch of documents
-    Note that the documents should already be batched at this point so that it does not inflate the
-    memory requirements"""
+    r = (0, 0)
+    try:
+        r = index_doc_batch(
+            chunker=chunker,
+            embedder=embedder,
+            document_index=document_index,
+            document_batch=document_batch,
+            index_attempt_metadata=index_attempt_metadata,
+            db_session=db_session,
+            ignore_time_skip=ignore_time_skip,
+        )
+    except Exception as e:
+        if index_attempt_metadata.num_exceptions < INDEXING_EXCEPTION_LOG_LIMIT:
+            trace = traceback.format_exc()
+            create_index_attempt_error(
+                attempt_id,
+                batch=index_attempt_metadata.batch_num,
+                docs=document_batch,
+                exception_msg=str(e),
+                exception_traceback=trace,
+                db_session=db_session,
+            )
+            logger.exception(
+                f"Indexing batch {index_attempt_metadata.batch_num} failed. msg='{e}' trace='{trace}'"
+            )
+
+        index_attempt_metadata.num_exceptions += 1
+        if index_attempt_metadata.num_exceptions >= INDEXING_EXCEPTION_LOG_LIMIT:
```

Comment:
thoughts on failing the indexing attempt if we see more than `INDEXING_EXCEPTION_LOG_LIMIT` failures?

This avoids silent failures (e.g. failed batches with no information given to the user)

Reply:
i think there's some different ways to head with this, but for the moment I've changed the var to INDEXING_EXCEPTION_LIMIT and am simply failing as suggested.  Note that the default is now 0, which maintains the fail instantly behavior prior to this PR.  In order to see the new changes the var will need to be adjusted to a non-zero number.  We can change the default behavior later once we get some feedback on how everything is working out.